src/interface.jl

```diff
@@ -0,0 +1,89 @@
+# All Geometries
+"""Returns the geometry type, such as `Polygon` or `Point`."""
+function geomtype(geom)
+    throw(ErrorException(string("Unknown Geometry type. ",
+    "Define GeoInterface.geomtype(::$(typeof(geom))) to return the desired type.")))
+end
+
+# All types
+"""`ncoord(geom) -> Integer`
+Return the number of coordinate dimensions (such as XYZ) for the geometry.````
+"""
+ncoord(geom) = ncoord(geomtype(geom), geom)
+
+"""Return `true` when the geometry is empty."""
+isempty(geom) = ngeom(geom) == 0
+"""Return `true` when the geometry doesn't cross itself."""
+issimple(geom) = issimple(geomtype(geom), geom)
+
+# Point
+getcoord(geom, i::Integer) = getcoord(geomtype(geom), geom, i)
+
+# Curve, LineString, MultiPoint
+npoint(geom) = npoint(geomtype(geom), geom)
+getpoint(geom, i::Integer) = getpoint(geomtype(geom), geom, i)
+
+# Curve
+startpoint(geom)
```

Comment:
This and more below is a function call, not a function definition. If there is no good fallback definition, we can also just declare an empty function, `function startpoint end`.